### PR TITLE
Handle Eventual Consistency Problem in Create Driver Call

### LINKED ISSUE #54: Handle the `Eventual Consistency' during Machine creation
**What happened**:
MCM triggers creation of multiple instances on AWS for the same backing machine object due to AWS' 'Eventual Consistency` implementation. 

**What you expected to happen**:
MCM should not create multiple instances for the same backing machine object. 

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know**:
This needs to be probably handled in `CreateMachine` method by introducing a `DescribeInstance()` method with an exponential backoff mechanism with a timeout of 2 minutes as suggested by AWS docs. This needs to be checked after creation of instance and before returning the response from `CreateMachine` method.

**Environment**:

### PR DESCRIPTION
**Pull request by himanshu-kun**
**What this PR does / why we need it**:
This PR takes care of problem where due to eventual consistency problem of AWS multiple instances get created
**Which issue(s) this PR fixes**:
Fixes #54 

**Special notes for your reviewer**:
I have made `maxElapsedTimeInBackoff` as a variable and not a `const` because I change this for the unit tests so that for unit-tests the retrying interval is 10sec and not 5min.
**Release note**:

```bugfix user
A problem is resolved where multiple VMs were created by the driver in case the APIs on the AWS are not consistent for some time.
```

### CODE REVIEW

**Review by AxiomSamarth — COMMENTED** [review]
/lgtm

File: pkg/aws/core_util.go [hunk 1]
```diff
@@ -159,6 +159,19 @@ func (d *Driver) getInstancesFromMachineName(machineName string, providerSpec *a
 	return instances, nil
 }
 
+func confirmInstanceById(svc ec2iface.EC2API, instanceID string) (bool,error){
```
Comment (AxiomSamarth) [inline pkg/aws/core_util.go:162]:
```suggestion
func confirmInstanceByID(svc ec2iface.EC2API, instanceID string) (bool,error){
```

**Review by AxiomSamarth — CHANGES_REQUESTED** [review]

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Review by dkistner — APPROVED** [review]
/lgtm

**Pull request merged**